### ISSUE TITLE
Validator: validate that no two constraints should have the same id

### ISSUE THREAD

**Issue by ewoutkramer** [FirelyTeam/firely-net-sdk#2215]
source is this discussion: https://jira.hl7.org/browse/FHIR-37692

**Comment by brianpos** [comment]
In the differential right? (as there definitely could be in the snapshot)

**Comment by ewoutkramer** [comment]
Yeah, there are a lot of exceptions, so it needs to be handwritten, as this would be too complex to formulate as FP.
